pkgs/development/compilers/cudatoolkit/stdenv.nix

```diff
@@ -1,17 +1,27 @@
-{ nixpkgsStdenv
-, nvccCompatibleStdenv
+{ nixpkgsCompatibleLibstdcxx
+, nvccCompatibleCC
 , overrideCC
+, stdenv
 , wrapCCWith
 }:
 
-overrideCC nixpkgsStdenv (wrapCCWith {
-  cc = nvccCompatibleStdenv.cc.cc;
+let
+  cc = wrapCCWith
+    {
+      cc = nvccCompatibleCC;
+
+      # This option is for clang's libcxx, but we (ab)use it for gcc's libstdc++.
+      # Note that libstdc++ maintains forward-compatibility: if we load a newer
+      # libstdc++ into the process, we can still use libraries built against an
+      # older libstdc++. This, in practice, means that we should use libstdc++ from
+      # the same stdenv that the rest of nixpkgs uses.
+      # We currently do not try to support anything other than gcc and linux.
+      libcxx = nixpkgsCompatibleLibstdcxx;
+    };
+  cudaStdenv = overrideCC stdenv cc;
```

Comment:
AFAICT this is the only usage of `cudaStdenv`. What's the intended difference b/w `cudaStdenv` and `backendStdenv`?

Reply:
I was hoping to rename `backendStdenv` into `cudaStdenv` some time later

Reply:
ah ok gotcha!